fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexValidator.java

```diff
@@ -56,6 +57,22 @@ public void validate(@Nonnull MetaDataValidator metaDataValidator) {
     public static void validateIndexOptions(@Nonnull Index index, @Nonnull RecordMetaData recordMetaData) {
         validateAnalyzerNamePerFieldOption(LuceneIndexOptions.LUCENE_ANALYZER_NAME_PER_FIELD_OPTION, index);
         validateAnalyzerNamePerFieldOption(LuceneIndexOptions.AUTO_COMPLETE_ANALYZER_NAME_PER_FIELD_OPTION, index);
+        validatePrimaryKeyOptions(index.getOptions());
+    }
+
+    private static void validatePrimaryKeyOptions(final Map<String, String> options) {
+        if (Boolean.parseBoolean(options.get(LuceneIndexOptions.PRIMARY_KEY_SEGMENT_INDEX_V2_ENABLED))) {
+            if (Boolean.parseBoolean(options.get(LuceneIndexOptions.PRIMARY_KEY_SEGMENT_INDEX_ENABLED))) {
+                throw new MetaDataException(
+                        "Index cannot enable both " + LuceneIndexOptions.PRIMARY_KEY_SEGMENT_INDEX_V2_ENABLED + " and "
+                                + LuceneIndexOptions.PRIMARY_KEY_SEGMENT_INDEX_ENABLED);
+            }
+            if (Boolean.parseBoolean(options.get(LuceneIndexOptions.OPTIMIZED_STORED_FIELDS_FORMAT_ENABLED))) {
```

Comment:
This means that if I had `optimizedStoredFieldsFormatEnabled=false` in the index options, it would be fine. But something like that suggests a (confused) attempt to disable `usesOptimizedStoredFields`, which it won't.